seonoh/ArchitectureStudy/app/src/main/java/com/example/seonoh/seonohapp/util/CalculateUtils.kt

```diff
@@ -25,95 +22,101 @@ object CalculateUtils {
+                        totalPriceAmount /= 1_000_000L
+                        mapValue["format"] = R.string.trade_amount_mega_fmt
+                        mapValue["price"] = totalPriceAmount
+                    }
+                    else -> {
+                        totalPriceAmount /= 1_000_000_000L
+                        mapValue["format"] = R.string.trade_amount_giga_fmt
+                        mapValue["price"] = totalPriceAmount
+
+                    }
+                }
             }
             "BTC", "ETH" -> {
-                String.format(context.getString(R.string.trade_amount_milli_fmt), accTradePrice24h)
+                mapValue["format"] = R.string.trade_amount_milli_fmt
+                mapValue["price"] = accTradePrice24h
             }
             "USDT" -> {
-                String.format(
-                    context.getString(
-                        when {
-                            totalPriceAmount < 1_000_000L -> {
-                                R.string.trade_amount_fmt
-                            }
-                            totalPriceAmount < 1_000_000_000L -> {
-                                totalPriceAmount /= 1_000L
-                                R.string.trade_amount_kilo_fmt
-                            }
-                            totalPriceAmount < 1_000_000_000_000L -> {
-                                totalPriceAmount /= 1_000_000L
-                                R.string.trade_amount_mega_fmt
-                            }
-                            else -> {
-                                totalPriceAmount /= 1_000_000_000L
-                                R.string.trade_amount_giga_fmt
-                            }
-                        }
-                    ), totalPriceAmount
-                )
+                when {
+                    totalPriceAmount < 1_000_000L -> {
+                        mapValue["format"] = R.string.trade_amount_fmt
+                        mapValue["price"] = totalPriceAmount
+                    }
+                    totalPriceAmount < 1_000_000_000L -> {
+                        totalPriceAmount /= 1_000L
+                        mapValue["format"] = R.string.trade_amount_kilo_fmt
+                        mapValue["price"] = totalPriceAmount
+                    }
+                    totalPriceAmount < 1_000_000_000_000L -> {
+                        totalPriceAmount /= 1_000_000L
+                        mapValue["format"] = R.string.trade_amount_mega_fmt
+                        mapValue["price"] = totalPriceAmount
+                    }
+                    else -> {
+                        totalPriceAmount /= 1_000_000_000L
+                        mapValue["format"] = R.string.trade_amount_giga_fmt
+                        mapValue["price"] = totalPriceAmount
+                    }
+                }
             }
             else -> {
                 val fmtResId = when {
```

Comment:
when 의 리턴 타입이 없습니다.

이와 같이 변경하여 처리하시는 것은 어떤가요?
```suggestion
                mapValue["format"] = when {
```

Reply:
감사합니다 Any -> Number로 변경 완료했습니다.

25a3b5d